pkg/util/net/net.go

```diff
@@ -72,3 +89,163 @@ func setBuffers(rc syscall.RawConn, sz int) error {
 
 	return err
 }
+
+func RemovePrivateDNSZone(
+	ctx context.Context,
+	log *logrus.Entry,
+	privateZonesClient privatedns.PrivateZonesClient,
+	configcli configclient.Interface,
+	mcocli mcoclient.Interface,
+	kubernetescli kubernetes.Interface,
+	vNetLinksClient privatedns.VirtualNetworkLinksClient,
+	resourceGroupID string) error {
+	if privateZonesClient == nil {
+		return errors.New("privateZonesClient is nil")
+	}
+
+	resourceGroup := stringutils.LastTokenByte(resourceGroupID, '/')
+	zones, err := privateZonesClient.ListByResourceGroup(ctx, resourceGroup, nil)
+	if err != nil {
+		log.Print(err)
+		return nil
+	}
+
+	if configcli == nil {
+		return errors.New("configcli is nil")
+	}
+
+	if len(zones) == 0 {
+		// fix up any clusters that we already upgraded
+		if err := UpdateDNSs(ctx, configcli.ConfigV1().DNSes(), resourceGroupID); err != nil {
+			log.Print(err)
+		}
+		return nil
+	}
+
+	if mcocli == nil {
+		return errors.New("mcocli is nil")
+	}
+
+	if kubernetescli == nil {
+		return errors.New("kubernetescli is nil")
+	}
+
+	mcpLister := mcocli.MachineconfigurationV1().MachineConfigPools()
+	nodeLister := kubernetescli.CoreV1().Nodes()
+	sameNumberOfNodesAndMachines, err := ready.SameNumberOfNodesAndMachines(ctx, mcpLister, nodeLister)
+	if err != nil {
+		log.Print(err)
+		return nil
+	}
+
+	if !sameNumberOfNodesAndMachines {
+		return nil
+	}
+
+	clusterVersionIsLessThan4_4, err := version.ClusterVersionIsLessThan4_4(ctx, configcli)
+	if err != nil {
+		log.Print(err)
+		return nil
+	}
+
+	if clusterVersionIsLessThan4_4 {
+		log.Printf("cluster version < 4.4, not removing private DNS zone")
+		return nil
+	}
+
+	if err = UpdateDNSs(ctx, configcli.ConfigV1().DNSes(), resourceGroupID); err != nil {
```

Comment:
nit: plural on `UpdateDNSs` does not match plural on `configcli.ConfigV1().DNSes()`

Reply:
Done.